lib/6to5/generation/buffer.js

```diff
@@ -67,32 +67,57 @@ Buffer.prototype.removeLast = function (cha) {
 
 Buffer.prototype.newline = function (i, removeLast) {
   if (this.format.compact) return;
-
-  if (_.isBoolean(i)) {
-    removeLast = i;
-    i = null;
-  }
+  removeLast = removeLast || false;
 
   if (_.isNumber(i)) {
     if (this.endsWith("{\n")) i--;
     if (this.endsWith(util.repeat(i, "\n"))) return;
 
-    for (var j = 0; j < i; j++) {
-      this.newline(null, removeLast);
+    while (i--) {
+      this._newline(removeLast);
     }
     return;
   }
 
+  if (_.isBoolean(i)) {
+    removeLast = i;
+  }
+
+  this._newline(removeLast);
+};
+
+Buffer.prototype._newline = function (removeLast) {
   if (removeLast && this.isLast("\n")) this.removeLast("\n");
 
   this.removeLast(" ");
 
   // remove whitespace if last character was a newline
-  this.buf = this.buf.replace(/\n +$/, "\n");
-
+  this._removeSpacesAfterLastNewline();
   this._push("\n");
 };
 
+/**
+ * If buffer ends with a newline and some spaces after it, trim those spaces.
+ */
+Buffer.prototype._removeSpacesAfterLastNewline = function () {
+  var lastNewlineIndex = this.buf.lastIndexOf('\n');
```

Comment:
It's hard to serve two masters. I think we should focus on improving perf for processing lots of small module-size files rather than a couple of big ones. It might be worth adding a console warning to the `6to5` binary and maybe some plugins that says we prefer 6to5 be run before concatenation or any other post-processing. It makes sense IMO. Webpack does this as well.